Formula/shogun.rb

```diff
@@ -0,0 +1,118 @@
+  sha256 "57169dc8c05b216771c567b2ee2988f14488dd13f7d191ebc9d0703bead4c9e6"
+
+  depends_on "cmake" => :build
+  depends_on :java => ["1.7+", :build]
+  depends_on "pkg-config" => :build
+  depends_on "swig" => :build
+  depends_on "arpack"
+  depends_on "eigen"
+  depends_on "glpk"
+  depends_on "hdf5"
+  depends_on "json-c"
+  depends_on "lapack"
+  depends_on "lzo"
+  depends_on "nlopt"
+  depends_on "python" if MacOS.version <= :lion
+  depends_on "readline"
+  depends_on "snappy"
+  depends_on "xz"
+
+  # fixes the linking of the python inteface
+  # based on the patch:
+  # https://github.com/shogun-toolbox/shogun/commit/ff8840ce0e0091d7619dacdf84819ea19872baf4
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/9df360c/shogun/fix_python_linking.patch"
+    sha256 "9c7b34b53ab6f54129c570c682b4a3fcae78418b211b7a8607db9b44d4daad35"
+  end
+
+  # fixes when Accelerator framework is trying to be used as a LAPACK backend for Eigen
+  # cmake swallows some of the include header flags hence on some versions of OSX
+  # the include of <vecLib/cblas.h> will fail
+  # based on the patch:
+  # https://github.com/shogun-toolbox/shogun/commit/6db834fb4ca9783b6e5adfde808d60ebfca0abc9
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/9df360c/shogun/fix_veclib.patch"
+    sha256 "de7ebe4c91da9f63fc322c5785f687c0005ed8df2c70cd3e9024fbac7b6d3745"
+  end
+
+  resource "jblas" do
+    url "https://mikiobraun.github.io/jblas/jars/jblas-1.2.3.jar"
+    sha256 "e9328d4e96db6b839abf50d72f63626b2309f207f35d0858724a6635742b8398"
+  end
+
+  resource "numpy" do
+    url "https://files.pythonhosted.org/packages/ee/66/7c2690141c520db08b6a6f852fa768f421b0b50683b7bbcd88ef51f33170/numpy-1.14.0.zip"
+    sha256 "3de643935b212307b420248018323a44ec51987a336d1d747c1322afc3c099fb"
+  end
+
+  needs :cxx11
+
+  def install
+    ENV.cxx11
+
+    # fix build of modular interfaces with SWIG 3.0.5 on MacOSX
+    # https://github.com/shogun-toolbox/shogun/pull/2694
+    # https://github.com/shogun-toolbox/shogun/commit/fef8937d215db7
+    ENV.append_to_cflags "-D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0"
+
+    ENV["PYTHONPATH"] = libexec/"lib/python2.7/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+
+    resource("numpy").stage do
+      system "python", *Language::Python.setup_install_args(libexec/"vendor")
+    end
+
+    system "python", *Language::Python.setup_install_args(libexec)
+    bin.install Dir[libexec/"bin/*"]
+    env = {
+      :PATH => "#{libexec}/vendor/bin:$PATH",
+      :PYTHONPATH => ENV["PYTHONPATH"],
+      :LAPACKE_PATH => '#{Formula["lapack"].opt_lib}',
+    }
+    bin.env_script_all_files(libexec/"bin", env)
+
+    libexec.install resource("jblas")
+
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args, "-DBUILD_EXAMPLES=OFF",
```

Comment:
```
      system "cmake", "..", "-DBUILD_EXAMPLES=OFF",
                            "-DBUNDLE_JSON=OFF",
                            "-DBUNDLE_NLOPT=OFF",
                            "-DENABLE_TESTING=OFF",
                            "-DENABLE_COVERAGE=OFF",
                            "-DBUILD_META_EXAMPLES=OFF",
                            "-DINTERFACE_PYTHON=ON",
                            "-DINTERFACE_JAVA=ON",
                            "-DJBLAS=#{libexec}/jblas-#{resource("jblas").version}.jar",
                            "-DLIB_INSTALL_DIR=#{lib}", 
                            *std_cmake_args
```

Reply:
``` "-DJBLAS=#{libexec}/j```

will run over the 80 char limit, not a problem?

Reply:
nope